oci/spec_opts_nonlinux.go

```diff
@@ -61,3 +63,16 @@ func WithRdt(closID, l3CacheSchema, memBwSchema string) SpecOpts {
 		return errors.New("RDT not supported")
 	}
 }
+
+func WithCDI(annotations map[string]string, cdiSpecDirs []string) SpecOpts {
+	return func(ctx context.Context, _ Client, c *containers.Container, s *Spec) error {
+		_, cdiDevices, err := cdi.ParseAnnotations(annotations)
+		if err != nil {
+			return fmt.Errorf("failed to parse CDI device annotations: %w", err)
+		}
+		if cdiDevices == nil {
+			return nil
+		}
+		return errors.New("CDI not supported")
+	}
+}
```

Comment:
Is there supposed to be a Linux version of this function being added too? Given that CDI itself is cross-platform, why behave differently for Linux and non-Linux?

I'm confused how this compiles on Linux at all.